src/ui/pages/work/activities/dialogs/AddActivityDialog.js

```diff
@@ -61,12 +61,28 @@ function AddActivityDialog(props) {
     };
 
     let updateFormValues = (key, value) => {
+        if (key === "taskCode") {
+            onTaskCodeChanged(value);
+        }
         setFormValues(prevFormValues => ({
             ...prevFormValues,
             [key]: value
         }))
     };
 
+    let onTaskCodeChanged = (taskcode) => {
+        setLoading(true);
+        WSWorkorders.getTaskPlan(taskcode).then(response => {
+            const taskPlan = response.body.data;
+            taskPlan.peopleRequired === null ? 
+                updateFormValues("peopleRequired", "") : updateFormValues("peopleRequired", taskPlan.peopleRequired);
+            taskPlan.estimatedHours === null ?
+                updateFormValues("estimatedHours", "") : updateFormValues("estimatedHours", taskPlan.estimatedHours);
+            taskPlan.tradeCode === null ?
+                updateFormValues("tradeCode", "") : updateFormValues("tradeCode", taskPlan.tradeCode);
```

Comment:
I think it would be better to change this to
```
updateFormValues("peopleRequired", taskPlan.peopleRequired === null ? "" : taskPlan.peopleRequired);
```
(and so on)